packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -263,18 +263,28 @@ class AndroidDevice extends Device {
   }
 
   bool _installLatestApp(ApplicationPackage package) {
-    if (isAppInstalled(package)) {
+    final bool wasInstalled = isAppInstalled(package);
+    if (wasInstalled) {
       if (isLatestBuildInstalled(package)) {
         printStatus('Latest build already installed.');
```

Comment:
We should probably make this a Trace as well. No need to tell the user this normally.